docs/specifications/tests/Methods.md

```diff
@@ -1,69 +1,280 @@
+         RDATA of the NS records in the answer section.
+      4. For each [in-bailiwick] name server, query the name server for A and
+         AAAA and extract the record in the answer or additional section.
+	 Follow any referal to sub-zone if needed.
+   6. For each [out-of-bailiwick] name servers do a normal recursive lookup 
+      for those to a resolver by querying for A and AAAA.
+   7. Collect all matching A and AAAA records from the responses.
+   8. If no IP addresses have been collected, return with ERROR.
+
+2. Undelegated test.
+
+   1. For every [in-bailiwick] name server in input data, collect IPv4 and
+      IPv6 address for the server name.
+   2. For all [out-of-bailiwick] name servers in input data, if it has 
+      neither IPv4 nor IPv6 data, do a normal recursive lookup for 
+      the address to a resolver by querying for A and AAAA. 
+   3. If a name server has either IPv4 or IPv6 data, do not query for
+      any data.
+   4. If no IP addresses have been collected, return with ERROR.
+
+
+### Outcome(s)
+
+Unless ERROR was returned in the steps, outcome is a set of IP addresses.
+
+### Special procedural requirements
+
+If a name server name points at a CNAME or DNAME follow that to extract
+an IP address, if possible. It is, however, not permitted for a NS record
+to point at a name that has a CNAME or DNAME, but that test i covered by
+Test Case [DELEGATION05].
+
+### Dependencies
+
+Test Case [BASIC01] must have been run.
+
+
+
+## Method 5: In-zone addresses records of name servers
+
+### Method identifier
+**METHOD5** In-zone addresses records of name servers
+
+
+### Objective
+
+Obtain the addresses of the [in-bailiwick] name servers, if any, for 
+the child zone as defined in the child zone itself. This method will
+ignore any addresses of [out-of-bailiwick] name servers.
+
+
+### Inputs
+
+* The name of the child zone ("child zone").
+* The fact if the child zone exists from [BASIC01].
+* The fact if the test type is undelegated test ("undelegate test") or not
+  ("normal test").
+* The addresses to the name servers of the child zone, as given by 
+  [METHOD4] ("name server IPs").
+* The name server names defined in the zone as defined by [METHOD2]
+  ("child zone name server names").
+* The fact if the test type is undelegated test ("undelegate test") or not
+  ("normal test").
+
+### Ordered description of steps to be taken to execute the method
+
+1. If the child zone does not exist, end with an ERROR unless the
+   test is *undelegated test*.
+2. For each *child zone name server name* that is an [in-bailiwick]
+   name server:
+   1. Send an A and an AAAA query to all servers in *name server IPs*.
+   2. If a delegation (referal) to a sub-zone of child zone is returned, 
+      follow that delegation, possibly in several steps, by repeating the
+      A and AAAA queries.
+   3. Ignore non-referal responses unless AA flag is set. Cached data
+      is not accepted.
```

Comment:
Please integrate the AA requirement in the previous step.
Is cached data considered everywhere unless it is explicitly ignored? Do we say somewhere how the cache is populated?

Reply:
Please elaborate why you think the current wording is not good.